bench-templates/src/macros/field.rs

```diff
@@ -1,542 +1,468 @@
+                        bytes.clear();
+                        f[i].serialize_uncompressed(&mut bytes).unwrap()
+                    })
+                },
+            );
+            serialization.bench_function(
+                &format!("Deserialize Compressed for {}", stringify!($F)),
+                |b| {
+                    let mut i = 0;
+                    b.iter(|| {
+                        i = (i + 1) % SAMPLES;
+                        bytes.clear();
+                        <$F>::deserialize_compressed(f_compressed[i].as_slice()).unwrap()
+                    })
+                },
+            );
+            serialization.bench_function(
+                &format!("Deserialize Compressed Unchecked for {}", stringify!($F)),
+                |b| {
+                    let mut i = 0;
+                    b.iter(|| {
+                        i = (i + 1) % SAMPLES;
+                        bytes.clear();
+                        <$F>::deserialize_compressed_unchecked(f_compressed[i].as_slice()).unwrap()
+                    })
+                },
+            );
+            serialization.bench_function(
+                &format!("Deserialize Uncompressed for {}", stringify!($F)),
+                |b| {
+                    let mut i = 0;
+                    b.iter(|| {
+                        i = (i + 1) % SAMPLES;
+                        bytes.clear();
+                        <$F>::deserialize_uncompressed(f_uncompressed[i].as_slice()).unwrap()
+                    })
+                },
+            );
+            serialization.bench_function(
+                &format!("Deserialize Uncompressed Unchecked for {}", stringify!($F)),
+                |b| {
+                    let mut i = 0;
+                    b.iter(|| {
+                        i = (i + 1) % SAMPLES;
+                        bytes.clear();
+                        <$F>::deserialize_uncompressed_unchecked(f_uncompressed[i].as_slice())
+                            .unwrap()
+                    })
+                },
+            );
+            serialization.finish()
         }
     };
 }
 
 #[macro_export]
 macro_rules! sqrt {
-    ($f:ident, $f_type:ty) => {
-        pub fn sqrt(b: &mut $crate::bencher::Bencher) {
+    ($F:ident) => {
+        fn sqrt(c: &mut $crate::criterion::Criterion) {
+            use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
             const SAMPLES: usize = 1000;
 
             let mut rng = ark_std::test_rng();
 
-            let v: Vec<$f_type> = (0..SAMPLES)
-                .map(|_| {
-                    let mut tmp = $f::rand(&mut rng);
-                    tmp.square_in_place();
-                    tmp
+            let f = (0..SAMPLES)
+                .map(|_| <$F>::rand(&mut rng))
+                .collect::<Vec<_>>();
+            let qrs = f.iter().map(|s| s.square()).collect::<Vec<_>>();
+            let qnrs = f.iter().map(|s| s.square()).collect::<Vec<_>>();
```

Comment:
Would `qnrs` be NRS?

Reply:
`qrs` = quadratic residues, and `qnrs` = quadratic non-residues. I think this is in line with our terminology elsewhere.

Reply:
The problem is that
```
 let qnrs = f.iter().map(|s| s.square()).collect::<Vec<_>>();
```
here qnrs are also qrs?

Reply:
I see. I was planning originally on making it `generator * s.square()`, but the generator doesn't yet exist for all fields. So for the time being I will remove it, as I don't know of another way to easily generate a non-square.

Reply:
Is it true that for a lot of the field we have access to a QNRS?

Reply:
Yeah, for Fp the generator is always a QNR, but for extension fields we don't necessarily have QNRs

Reply:
I feel maybe one day we will add a mandatory QNR field for extension fields. Thoughts?

Reply:
This is related to #276

Reply:
Some of the extension fields used to have QNRs which were not used anywhere, see https://github.com/arkworks-rs/algebra/pull/421/.

Reply:
Hm good catch, still not sure how we could use this generically.